### PR TITLE
Use ES6 spread syntax

### PR DESCRIPTION
**Pull request by juj**
Use ES6 spread syntax to micro-optimize code size (and perhaps perf) on some function calls.

### CODE REVIEW

File: src/library_html5.js [hunk 1]
```diff
@@ -105,7 +105,11 @@ var LibraryJSEvents = {
         var call = JSEvents.deferredCalls[i];
         JSEvents.deferredCalls.splice(i, 1);
         --i;
+#if MIN_CHROME_VERSION < 46 || MIN_FIREFOX_VERSION < 27 || MIN_IE_VERSION != TARGET_NOT_SUPPORTED || MIN_SAFARI_VERSION < 80000 // https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Spread_syntax#Spread_in_function_calls
```
Comment (sbc100) [inline src/library_html5.js]: Wouldn't it make sense to define this centrally?

This brings up the question again in mind of weather we should just switch to ES6 across all our internal code and rely on babel or some other transpiler when shipping to ES5 browsers.   As somebody who ships to a lot of older targets I wonder what you think about this option @juj ?

**Comment by kripken** [comment]
Another option is to write an optimization pass for this in the acorn JS optimizer. Then we'd just write a normal `apply` in JS but optimize it later. That might be risky on user code though.

File: src/library_html5.js (hunk 1 above)
Comment (juj) [reply src/library_html5.js]: Using babel will be an overkill. There are very few places in code that are ES6 that adding babel machinery to the build will not help these few lines. Also with babel we are no longer in control of what kind of code gets emitted, and looking at what it emits for [emulating spread syntax](https://stackoverflow.com/questions/55130183/how-do-i-detect-fallback-if-the-browser-has-no-spread-syntax-notation-in-fu), it looks pretty bad.

**Comment by juj** [comment]
> How about using `{{{ }}}` for this, though

> ```js
> {{{ makeApply('call.targetFunction', 'call.argsList') }}}
> ```

If we do want to generalize, I think it would be better to generalize from another perspective, i.e.

```js
#if BUILD_TARGET_SUPPORTS('es6-spread-syntax-in-function-calls')
        call.targetFunction(...call.argsList);
#else
        call.targetFunction.apply(null, call.argsList);
#endif
```
rather than add a special syntax for codegenning the call itself. That way we could have a central place with different feature literals that can be checked.

However in general I don't think there is much need to start building abstractions for this, since we have relatively little amount of library code that hand-maintaining the permutations is not much of a problem. (it becomes a problem if we have lots of nested #ifs, but I don't think that happens here)

**Comment by kripken** [comment]
I like `BUILD_TARGET_SUPPORTS` but I think it can be used with the `{{{ }}}` idea (internally in the called utility). I feel that 5 lines for each such instance is worse overall code - it's longer and it's less clear than a single line that explicitly says "make a call of a JS function with an array of arguments".

In particular, 2 of those 5 lines have duplications between them, so there is a risk of bugs. But really it's just clarity that I'm trying to maximize - we already have lots of ifdefs that are hard to avoid, but this one seems unnecessary.

**Comment by juj** [comment]
Closing as stale - I'll figure this one out later when I get time.

**Pull request closed without merging**